File: sorting/insertion_sort.py
def insertion_sort(customList):
    for i in range(1, len(customList)):
        key = customList[i]
        j = i - 1
        while j >= 0 and key < customList[j]:
            customList[j + 1] = customList[j]
            j -= 1
        customList[j + 1] = key
    print(customList)


list1 = [2, 1, 7, 6, 5, 3, 4, 8]
insertion_sort(list1)

File: sorting/insertionSort.js
function insertionSort(arr) {
  for (let i = 1; i < arr.length; i++) {
    let key = arr[i];
    let j = i - 1;
    while (j >= 0 && key < arr[j]) {
      arr[j + 1] = arr[j];
      j -= 1;
    }
    arr[j + 1] = key;
  }
  return arr;
}

let list1 = [2, 1, 7, 6, 5, 3, 4, 8];
console.log(insertionSort(list1));
